templates/aws-vpc.template.yaml

```diff
@@ -303,849 +345,911 @@ Parameters:
             Value: !Select
-              - '1'
+              - "1"
               - !Split
-                - '='
-                - !Ref 'PublicSubnetTag3'
-          - !Ref 'AWS::NoValue'
+                - "="
+                - !Ref "PublicSubnetTag3"
+          - !Ref "AWS::NoValue"
       MapPublicIpOnLaunch: true
   PublicSubnet4:
-    Condition: 4AZCondition
+    Condition: PublicSubnets&4AZCondition
     Type: AWS::EC2::Subnet
     Properties:
-      VpcId: !Ref 'VPC'
-      CidrBlock: !Ref 'PublicSubnet4CIDR'
+      VpcId: !Ref "VPC"
+      CidrBlock: !Ref "PublicSubnet4CIDR"
       AvailabilityZone: !Select
-        - '3'
-        - !Ref 'AvailabilityZones'
+        - "3"
+        - !Ref "AvailabilityZones"
       Tags:
         - Key: Name
           Value: Public subnet 4
         - !If
           - PublicSubnetTag1Condition
           - Key: !Select
-              - '0'
+              - "0"
               - !Split
-                - '='
-                - !Ref 'PublicSubnetTag1'
+                - "="
+                - !Ref "PublicSubnetTag1"
             Value: !Select
-              - '1'
+              - "1"
               - !Split
-                - '='
-                - !Ref 'PublicSubnetTag1'
-          - !Ref 'AWS::NoValue'
+                - "="
+                - !Ref "PublicSubnetTag1"
+          - !Ref "AWS::NoValue"
         - !If
           - PublicSubnetTag2Condition
           - Key: !Select
-              - '0'
+              - "0"
               - !Split
-                - '='
-                - !Ref 'PublicSubnetTag2'
+                - "="
+                - !Ref "PublicSubnetTag2"
             Value: !Select
-              - '1'
+              - "1"
               - !Split
-                - '='
-                - !Ref 'PublicSubnetTag2'
-          - !Ref 'AWS::NoValue'
+                - "="
+                - !Ref "PublicSubnetTag2"
+          - !Ref "AWS::NoValue"
         - !If
           - PublicSubnetTag3Condition
           - Key: !Select
-              - '0'
+              - "0"
               - !Split
-                - '='
-                - !Ref 'PublicSubnetTag3'
+                - "="
+                - !Ref "PublicSubnetTag3"
             Value: !Select
-              - '1'
+              - "1"
               - !Split
-                - '='
-                - !Ref 'PublicSubnetTag3'
-          - !Ref 'AWS::NoValue'
+                - "="
+                - !Ref "PublicSubnetTag3"
+          - !Ref "AWS::NoValue"
       MapPublicIpOnLaunch: true
   PrivateSubnet1ARouteTable:
     Condition: PrivateSubnetsCondition
     Type: AWS::EC2::RouteTable
     Properties:
-      VpcId: !Ref 'VPC'
+      VpcId: !Ref "VPC"
       Tags:
         - Key: Name
           Value: Private subnet 1A
         - Key: Network
           Value: Private
   PrivateSubnet1ARoute:
-    Condition: PrivateSubnetsCondition
+    Condition: PrivateSubnets&PublicSubnets&NatGatewaysCondition
```

Comment:
You can just use NATgatewaytrue condition, as both private and public subnet will be true due to assertion, when natgatewaycondition is true.